app/src/test/java/de/westnordost/streetcomplete/quests/shop_type/CheckShopTypeTest.kt

```diff
@@ -136,20 +136,26 @@ class CheckShopTypeTest {
         )
     }
 
-    // see KEYS_THAT_SHOULD_NOT_BE_REMOVED_WHEN_SHOP_IS_REPLACED
-    @Test fun `apply shop with tags answer clears all tags except for a few ones`() {
+    // see KEYS_THAT_SHOULD_BE_REMOVED_WHEN_SHOP_IS_REPLACED
+    @Test fun `apply shop with tags answer clears only specified known safe-to-remove tags`() {
         questType.verifyAnswer(
             mapOf(
                 "building" to "yes", // <- should not be cleared
                 "disused:amenity" to "yes",
                 "phone" to "123456",
+		"ref" to "1111",
+		"fee" to "yes",
+		"nycdoitt:bin" to "22222",
+		"barrier" to "wall",
                 "name" to "Juppiebude"
             ),
             ShopType(mapOf("shop" to "ice_cream")),
             StringMapEntryAdd("shop", "ice_cream"),
             StringMapEntryDelete("disused:amenity", "yes"),
             StringMapEntryDelete("phone", "123456"),
             StringMapEntryDelete("name", "Juppiebude"),
+            StringMapEntryDelete("ref", "1111"),
+            StringMapEntryDelete("fee", "yes"),
```

Comment:
From a purely Kotlin perspective (ie, I have not checked whether these tags are the correct ones that should be removed in this situation, this appears to correctly add 4 tags to the initial object, and expect that 2 of them (`ref` and `fee`) should be deleted. I believe it also checks that the other two are not deleted, although I'm not familiar with this assertions library and I didn't check the docs/source of `containsExactlyInAnyOrder` to be sure.

https://github.com/streetcomplete/StreetComplete/blob/c463be9ea67fa2153fadc072b740048a87e28641/app/src/test/java/de/westnordost/streetcomplete/quests/OsmElementQuestType.kt#L13

The tags added have a different indentation than the others, though.

Reply:
thanks! converted tabs to spaces now